### PR TITLE
Cart rework

### PR DESCRIPTION
**Pull request by charm-13**
carts.py, catalog.py -> shop.py
/shop/catalog: allows users to view the cafe's catalog of available treats
/shop/purchase: allows users to purchase a treat from the cafe

### CODE REVIEW

File: src/api/shop.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                        """
+                    ),
+                    {
+                        "order_id": purchase.order_id,
+                        "user_id": purchase.user_id,
+                        "treat_sku": purchase.treat_sku,
+                        "quantity": purchase.quantity,
+                    },
+                )
+                .mappings()
```
Comment (thuang-28) [inline src/api/shop.py]: `success` isn't used for anything but checking for existence, so you can probably leave out the `.mappings().fetchone()` entirely

**Review by thuang-28 — CHANGES_REQUESTED** [review]
looks very nice; the HTTP status codes look appropriate too. APISpec should be updated to reflect the rework

**Review by samikshakarimbil — APPROVED** [review]
LGTM

File: APISpec.md [hunk 2]
```diff
@@ -2,13 +2,7 @@
...
 ## 1. User Purchasing
 
-The API calls are made in this sequence when making a purchase:
-1. `Get Catalog`
-2. `New Cart`
-3. `Add Item to Cart` (Can be called multiple times)
-4. `Checkout Cart`
-
-### 1.1. Get Catalog - `/catalog/` (GET)
+### 1.1. Get Catalog - `shop/catalog/` (GET)
```
Comment (thuang-28) [inline APISpec.md]: missing `/` at beginning of endpoint URL

File: APISpec.md [hunk 3]
```diff
@@ -34,26 +28,8 @@ Creates a new cart for a specific user.
...
-### 1.3. Add Item to Cart - `/carts/{cart_id}/items/{item_sku}` (PUT)
-
-Updates the quantity of a specific item in a cart. 
-
-**Request**:
-
-```json
-{
+  "order_id": "integer",
+  "treak_sku": "string",
```
Comment (thuang-28) [inline APISpec.md]: typo...

**Review by thuang-28 — CHANGES_REQUESTED** [review]

File: APISpec.md (hunk 3 above)
Comment (charm-13) [reply APISpec.md]: lol whoopsies

**Review by thuang-28 — APPROVED** [review]

**Pull request merged**